state/container.go

```diff
@@ -42,64 +41,66 @@ type container struct {
 	dataStore     domain.StateDataMap               // Handler's container-specific storage blob
 	initProc      domain.ProcessIface               // container's init process
 	service       domain.ContainerStateServiceIface // backpointer to service
+	intLock       sync.RWMutex                      // internal lock
```

Comment:
The answer is "no", meaning that we could do with the external lock only. But I think removing the internal lock is error prone because it means that all users of the container type must now explicitly lock. This makes it harder to maintain in my view.

Reply:
Yes, i agree that 1. may be less error-prone with the current complexity (low). But if we see an upward trend in concurrency-handling logic, we should consider switching to message-passing approach so that we can sleep at night. Let's go with this for now.